src/devices_models/devices/common/add_to_expression.jl

```diff
@@ -196,6 +196,34 @@ function add_to_expression!(
     return
 end
 
+function add_to_expression!(
```

Comment:
it would be a union of AbstractCompactUnitCommitment and ThermalCompactDispatch, but no other thermal formulation tries to add On variable in the demand balance equations. So I just left it at AbstractThermalFormulation